src/cdk/observers/observe-content.ts

```diff
@@ -35,6 +32,70 @@ export class MutationObserverFactory {
   }
 }
 
+
+/** A factory that creates ContentObservers. */
+@Injectable({providedIn: 'root'})
+export class ContentObserver {
+  private _observedElements = new Map<Element, {
+    observer: MutationObserver | null,
+    stream: Subject<MutationRecord[]>,
+    count: number
+  }>();
+
+  constructor(private _mutationObserverFactory: MutationObserverFactory) {}
+
+  observe(element: Element, debounce?: number): Observable<MutationRecord[]> {
+    return Observable.create(observer => {
+      const stream = this._observeElement(element);
+      const subscription =
+          (debounce ? stream.pipe(debounceTime(debounce)) : stream).subscribe(observer);
+
+      return () => {
+        subscription.unsubscribe();
+        this._unobserveElement(element);
+      }
+    });
+  }
+
+  private _observeElement(element: Element): Subject<MutationRecord[]> {
+    if (!this._observedElements.has(element)) {
+      const stream = new Subject<MutationRecord[]>();
+      const observer = this._mutationObserverFactory.create(mutations => stream.next(mutations));
+      if (observer) {
+        observer.observe(element, {
+          characterData: true,
+          childList: true,
+          subtree: true
+        });
+      }
+      this._observedElements.set(element, {observer, stream, count: 1});
+    } else {
+      this._observedElements.get(element)!.count++;
+    }
+    return this._observedElements.get(element)!.stream;
+  }
+
+  private _unobserveElement(element: Element) {
+    if (this._observedElements.has(element)) {
+      if (!--this._observedElements.get(element)!.count) {
```

Comment:
Could you break up the decrement and the conditional? It's generally easier to understand that way, since many people have a hard time remembering the different between `--value` and `value--`